nav2_map_server/CMakeLists.txt

```diff
@@ -162,5 +170,7 @@ ament_export_libraries(
   ${library_name}
   ${map_io_library_name}
 )
-ament_export_dependencies(${map_io_dependencies} ${map_server_dependencies} yaml-cpp)
+ament_export_dependencies(nav_msgs nav2_msgs nav2_util rclcpp rclcpp_lifecycle)
```

Comment:
yaml-cpp? I think a user added that because they needed that exported 

tf2, rclcpp_components?

Reply:
`yaml-cpp`, `tf2`, and `rclcpp_components` do not need to be exported, because they are PRIVATE dependencies.  That is, downstream consumers do not need to know about the dependency in order to successfully build/link.